packages/agreement-process/src/services/readmodel/readModelService.ts

```diff
@@ -97,9 +117,7 @@ const getAgreementsFilters = (
         (agreementStates) => agreementStates.length > 0 && showOnlyUpgradeable
       ),
       (agreementStates) =>
-        upgradeableStates.filter(
-          (s1) => agreementStates.some((s2) => s1 === s2) !== undefined
-        )
+        upgradeableStates.filter((s) => agreementStates.includes(s))
```

Comment:
Also this one fixes a bug, the mix of agreementStates and showOnlyUpgradeable filters was not working properly